### ISSUE TITLE
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.Comparable.compareTo(java.lang.Object)' on a null object reference

### ISSUE THREAD

**Issue by superdiazzz** [jhy/jsoup#1397]
Hello Jsoup Team,

I am using this library with dependency is 'org.jsoup:jsoup:1.10.1'

My console log said 
```
06-03 20:28:34.124: W/System.err(25097): java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.Comparable.compareTo(java.lang.Object)' on a null object reference
06-03 20:28:34.124: W/System.err(25097): 	at java.util.Arrays.binarySearch0(Arrays.java:2367)
06-03 20:28:34.124: W/System.err(25097): 	at java.util.Arrays.binarySearch(Arrays.java:2307)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.nodes.Entities$EscapeMode.a(Entities.java:4)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.nodes.Entities.a(Entities.java:2)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.Tokeniser.a(Tokeniser.java:42)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.TokeniserState.b(TokeniserState.java:16)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.TokeniserState.a(TokeniserState.java:1)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.TokeniserState$2.a(TokeniserState.java:1)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.Tokeniser.j(Tokeniser.java:5)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.TreeBuilder.c(TreeBuilder.java:1)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.TreeBuilder.b(TreeBuilder.java:2)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.HtmlTreeBuilder.b(HtmlTreeBuilder.java:4)
06-03 20:28:34.124: W/System.err(25097): 	at org.jsoup.parser.Parser.b(Parser.java:2)
06-03 20:28:34.125: W/System.err(25097): 	at org.jsoup.Jsoup.a(Jsoup.java:2)
06-03 20:28:34.125: W/System.err(25097): 	at com.tribunnews.app.widget.b.a(ContentConjuction.kt:19)
06-03 20:28:34.125: W/System.err(25097): 	at com.tribunnews.app.widget.b.a(ContentConjuction.kt:2)
06-03 20:28:34.125: W/System.err(25097): 	at com.tribunnews.app.DetailActivity$d.a(DetailActivity.kt:37)
06-03 20:28:34.125: W/System.err(25097): 	at com.tribunnews.app.DetailActivity$d.a(DetailActivity.kt:1)
06-03 20:28:34.125: W/System.err(25097): 	at androidx.lifecycle.LiveData.b(LiveData.java:6)
06-03 20:28:34.125: W/System.err(25097): 	at androidx.lifecycle.LiveData.a(LiveData.java:8)
06-03 20:28:34.125: W/System.err(25097): 	at androidx.lifecycle.LiveData.b(LiveData.java:14)
06-03 20:28:34.125: W/System.err(25097): 	at androidx.lifecycle.t.b(MutableLiveData.java:1)
06-03 20:28:34.125: W/System.err(25097): 	at androidx.lifecycle.LiveData$a.run(LiveData.java:5)
06-03 20:28:34.125: W/System.err(25097): 	at android.os.Handler.handleCallback(Handler.java:873)
06-03 20:28:34.125: W/System.err(25097): 	at android.os.Handler.dispatchMessage(Handler.java:99)
06-03 20:28:34.125: W/System.err(25097): 	at android.os.Looper.loop(Looper.java:214)
06-03 20:28:34.125: W/System.err(25097): 	at android.app.ActivityThread.main(ActivityThread.java:6981)
06-03 20:28:34.125: W/System.err(25097): 	at java.lang.reflect.Method.invoke(Native Method)
06-03 20:28:34.125: W/System.err(25097): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
06-03 20:28:34.125: W/System.err(25097): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
```
Very hard to understand what causing this error, but when i check my code what i did like below:

```
fun parseArticleP(article: String): MutableList<ContentArticle?> {
        val contents: MutableList<ContentArticle?> = ArrayList()
        val doc = Jsoup.parse(article)
        // wrap FIGURE
        val elFigure = doc.select("figure")
        for (i in elFigure.indices) {
            val element = elFigure[i]
            element.wrap("<p></p>")
        }
        val elP = doc.select("p")
        for (el in elP) {
            val v = el.text()
            val p = el.outerHtml()
            val ca = ContentArticle()
            ca.setValue(v)
            ca.setP(p)
            contents.add(ca)
        }
        return contents
    }
```

First of all, i want to wraping all `figure` tag with `<p>`, then i will choose all tag `<p>` and put those all into arraylist object. `p` means all content include with tag, `v` means only value of tag. There is problem with my snippet code? why i got this error?

**Comment by krystiangorecki** [comment]
1. Is the exception also thrown when you remove all code and leave only this line?
```
 val doc = Jsoup.parse(article)
```
2. We're missing your `article` you're trying to parse. It may be difficult to guess without it.

**Comment by superdiazzz** [comment]
i will make the new test tomorrow due to my Firebase Test Lab only allowed to test 5 devices per day. This problem only occur in specific device (i have been tested with all my physically device, everything is okay, no issue) but when i saw the result by test lab in some device like pie and oreo (samsung). Test lab displaying white text for parse result. I have been tested with my level 28 Pixel emulator and it was good too.

**Comment by superdiazzz** [comment]
@krystiangorecki  i have tested  again with only single line 

![Screen Shot 2020-06-05 at 10 45 40](https://user-images.githubusercontent.com/10770867/83834953-bc190a00-a719-11ea-8e09-49441e4c5520.png)



the error occur again, what console print like below

```
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.Comparable.compareTo(java.lang.Object)' on a null object reference
     FATAL EXCEPTION: main
Process: mobile.android.tribun, PID: 32406
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.Comparable.compareTo(java.lang.Object)' on a null object reference
	at java.util.Arrays.binarySearch0(Arrays.java:2367)
	at java.util.Arrays.binarySearch(Arrays.java:2307)
	at org.jsoup.nodes.Entities$EscapeMode.a(Entities.java:4)
	at org.jsoup.nodes.Entities.a(Entities.java:2)
	at org.jsoup.parser.Tokeniser.a(Tokeniser.java:42)
	at org.jsoup.parser.TokeniserState$38.a(TokeniserState.java:8)
	at org.jsoup.parser.Tokeniser.j(Tokeniser.java:5)
	at org.jsoup.parser.TreeBuilder.c(TreeBuilder.java:1)
	at org.jsoup.parser.TreeBuilder.b(TreeBuilder.java:2)
	at org.jsoup.parser.HtmlTreeBuilder.b(HtmlTreeBuilder.java:4)
	at org.jsoup.parser.Parser.b(Parser.java:2)
	at org.jsoup.Jsoup.a(Jsoup.java:2)
	at com.tribunnews.app.widget.b.a(ContentConjuction.kt:19)
	at com.tribunnews.app.widget.b.a(ContentConjuction.kt:2)
	at com.tribunnews.app.DetailActivity$d.a(DetailActivity.kt:37)
	at com.tribunnews.app.DetailActivity$d.a(DetailActivity.kt:1)
	at androidx.lifecycle.LiveData.b(LiveData.java:6)
	at androidx.lifecycle.LiveData.a(LiveData.java:8)
	at androidx.lifecycle.LiveData.b(LiveData.java:14)
	at androidx.lifecycle.t.b(MutableLiveData.java:1)
	at androidx.lifecycle.LiveData$a.run(LiveData.java:5)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7032)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```

**Comment by superdiazzz** [comment]
@krystiangorecki would you like to help me? any requirement you need to inspection

**Comment by krystiangorecki** [comment]
I can't reproduce it on Windows. It parses correctly.

NPE originates from: `at org.jsoup.nodes.Entities$EscapeMode.a(Entities.java:4)`
and here's the code
```
    public enum EscapeMode {
        /** Restricted entities suitable for XHTML output: lt, gt, amp, and quot only. */
        xhtml("entities-xhtml.properties", 4),
        /** Default HTML output entities. */
        base("entities-base.properties", 106),
```
Looks like a problem with loading properties files as Android resources fixed in #959. It was fixed in 1.11.1 so try using at least this version. You're using a version that's over 3 years old!

btw Please never place you code as an image.

**Comment by superdiazzz** [comment]
@krystiangorecki i tried `implementation 'org.jsoup:jsoup:1.13.1` and did some test lab. Looks ok, there were no crash anymore. So far, everything is good. Thank you brooo
